lib/project_types/script/layers/domain/metadata.rb

```diff
@@ -38,7 +39,13 @@ def create_from_json(ctx, metadata_json)
               raise ::Script::Layers::Domain::Errors::MetadataValidationError, ctx.message(err_msg)
             end
 
-            Metadata.new(schema_major_version, schema_minor_version)
+            use_msgpack = if metadata_hash["flags"]
+              metadata_hash["flags"]["use_msgpack"]
+            else
+              false
+            end
```

Comment:
nit: if a flag other than `use_msgpack` is set, then this variable would be nil. Should we enforce a boolean? Maybe something like:

```suggestion
            use_msgpack = !!metadata_hash.dig("flags", "use_msgpack")
```